__tests__/unit/client.test.ts

```diff
@@ -33,26 +37,26 @@ describe("Client", () => {
   });
 
   it("Allows for creation and usage of a new client after first client closed.", async () => {
-    const nodeClient = NodeHTTP2Client.getClient();
+    const nodeClient = new NodeHTTP2Client(getDefaultHTTPClientOptions());
     const clientOne = new Client(getDefaultSecretAndEndpoint(), nodeClient);
-    expect(nodeClient.isClosed()).toBe(false);
     await clientOne.query(fql`"Hello World"`);
+    expect(nodeClient.isClosed()).toBe(false);
     clientOne.close();
     expect(nodeClient.isClosed()).toBe(true);
     const clientTwo = new Client(
       getDefaultSecretAndEndpoint(),
-      NodeHTTP2Client.getClient()
+      new NodeHTTP2Client(getDefaultHTTPClientOptions())
     );
-    expect(nodeClient.isClosed()).toBe(false);
     expect((await clientTwo.query(fql`"Hello World"`)).data).toEqual(
       "Hello World"
     );
+    expect(nodeClient.isClosed()).toBe(false);
     const clientThree = new Client(
       getDefaultSecretAndEndpoint(),
-      NodeHTTP2Client.getClient()
+      new NodeHTTP2Client(getDefaultHTTPClientOptions())
     );
     clientTwo.close();
-    expect(nodeClient.isClosed()).toBe(false);
+    expect(nodeClient.isClosed()).toBe(true);
```

Comment:
There isn't ref counting anymore. We can reimplement it, but I took it out because I was struggling to make it correct.

I ran with this because I think it is safe (if not most efficient) for this iteration as clients with different connection properties (url and session idle time) will use completely different sessions. Any client that has finds itself with a close session will create a new one. I also think there were issues with the ref counting before, for example

- all sessions get closed. numberOfUsers == 0.
- the first client to start back up gets the session from the map, sees it's closed, replaces with new session. numberOfUsers == 0
- the next client to make a requests get the latest session,m which is open.
- one of the clients calls `closed` numberOfUsers is 0 so session.close is called